server/src/main/java/org/elasticsearch/search/fetch/subphase/FetchSourceContext.java

```diff
@@ -89,10 +90,14 @@ public static FetchSourceContext readFrom(StreamInput in) throws IOException {
         final boolean fetchSource = in.readBoolean();
         final String[] includes = in.readStringArray();
         final String[] excludes = in.readStringArray();
+        if (in.getTransportVersion().onOrAfter(TransportVersions.HIDE_VECTORS_FROM_SOURCE)) {
+            final Boolean includeVectors = in.readOptionalBoolean();
+            return of(fetchSource, includes, excludes, includeVectors);
+        }
         return of(fetchSource, includes, excludes);
```

Comment:
Don't return early here. Instead, set `includeVectors` as appropriate:

```suggestion
        final Boolean includeVectors;
        if (in.getTransportVersion().onOrAfter(TransportVersions.HIDE_VECTORS_FROM_SOURCE)) {
            includeVectors = in.readOptionalBoolean();
        } else {
            includeVectors = true;
        }
        return of(fetchSource, includes, excludes, includeVectors);
```

Reply:
Make sense, this will help to handle backward compatibility.